utils/python/CIME/build.py

```diff
@@ -508,43 +508,54 @@ def _build_model_thread(config_dir, compclass, caseroot, libroot, bldroot, incro
     for mod_file in glob.glob(os.path.join(bldroot, "*_[Cc][Oo][Mm][Pp]_*.mod")):
         shutil.copy(mod_file, incroot)
 
+def cleantree(base, ignoredirs=None):
+    """
+    clean all files below base level, leave directories and top level files
+    """
+    ignoredirs.append(base)
+    for root, _, files in os.walk(base, topdown=False):
+        if root not in ignoredirs:
+            for name in files:
+                os.remove(os.path.join(root, name))
+
+
 ###############################################################################
-def clean(case, cleanlist=None):
+def clean(case, cleanlist=None, clean_all=False):
 ###############################################################################
+    caseroot = case.get_value("CASEROOT")
+    if clean_all:
+        # If cleanlist is empty just remove the bld directory
+        exeroot = case.get_value("EXEROOT")
+        if os.path.isdir(exeroot):
+            logging.info("cleaning directory %s" %exeroot)
+            cleantree(exeroot, ignoredirs=[os.path.join(exeroot,"lib")])
```

Comment:
This was part of trying to do --clean by just removing the directories.   I started by removing the bld directory but then several tests failed because they were expecting to find dependencies in bld/lib so then I went back to using make to do the clean operation.   Since this code is only for clean-all now I think I can go back to rmtree and just completely remove bld.